spring-cloud-deployer-spi/src/main/java/org/springframework/cloud/deployer/spi/app/AppDeployer.java

```diff
@@ -135,4 +137,13 @@
 	 * @return the app deployment status
 	 */
 	AppStatus status(String id);
+
+	/**
+	 * Return the environment info for this deployer. This info should include the
+	 * following: 'deployer-class-name','deployer-version', 'deployer-spi-version', 'platform-type',
+	 * 'platform-client-version', 'platform-host-version', 'java-version', 'spring-version', 'spring-boot-version'.
+	 *
+	 * @return the app deployment environment info
+	 */
+	Map<String, String>  environmentInfo();
```

Comment:
I'd +1.

Note that `deployer-classname` is something that 
1) is not very interesting to see for an end-user once you have the other properties
2) can be retrieved by doing `getClass()` on the very object implementing this interface